pkg/client/unversioned/configdata.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unversioned
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+type ConfigDatasNamespacer interface {
+	ConfigDatas(namespace string) ConfigDatasInterface
+}
+
+type ConfigDatasInterface interface {
+	Get(string) (*extensions.ConfigData, error)
+	List(labels.Selector, fields.Selector) (*extensions.ConfigDataList, error)
+	Create(*extensions.ConfigData) (*extensions.ConfigData, error)
+	Delete(string) error
+	Update(*extensions.ConfigData) (*extensions.ConfigData, error)
+	UpdateStatus(*extensions.ConfigData) (*extensions.ConfigData, error)
+	Watch(labels.Selector, fields.Selector, string) (watch.Interface, error)
+}
+
+type configDatas struct {
+	client    *Client
+	namespace string
+}
+
+func newConfigDatas(c *Client, ns string) *configDatas {
+	return &configDatas{
+		client:    c,
+		namespace: ns,
+	}
+}
+
+func (cfg *configDatas) resourceName() string {
```

Comment:
I'm guessing this is a function so you don't have a magic constant littering the code.  This is a good thing to want to do, but make it a package-scoped const named something like `configDataResourceName`.